include/bolgenos-ng/type_traits.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+
+template<class ValueType, ValueType Value>
+struct integral_constant {
```

Comment:
:exclamation: move to `lib/` folder and `lib::` namespaces since it STL-like functionality